test.js

```diff
@@ -91,3 +92,40 @@ test('kill from port', async t => {
 test('error when process is not found', async t => {
 	await t.throwsAsync(fkill(['notFoundProcess']), /Killing process notFoundProcess failed: Process doesn't exist/);
 });
+
+function testKillDescendant(t, name = '') {
+	const fixture = path.resolve('fixtures', 'descendant');
+	const cp = childProcess.spawn('node', [fixture, name]);
+
+	cp.stdout.setEncoding('utf8');
+	cp.stdout.on('data', async chunk => {
+		const descendantPid = parseInt(chunk, 10);
+		t.is(typeof descendantPid, 'number');
+
+		const opts = {
+			tree: true,
+			force: process.platform === 'win32'
+		};
+
+		if (name) {
+			await fkill(name, opts);
+		} else {
+			await fkill(cp.pid, opts);
+		}
+
+		// Ensure all the processes has time to exit
+		await delay(400);
+
+		t.false(await processExists(cp.pid));
+		t.false(await processExists(descendantPid));
+		t.end();
+	});
+}
+
+// eslint-disable-next-line ava/test-ended
```

Comment:
I've disabled eslint for those line because the rule `ava/test-ended` seems to not correctly report the usage of `t.end()` when using a macro. Later on I may report as an issue to `ava` repo.